### PR TITLE
#27878 removing the lock to look for on the cache

### LINKED ISSUE #27878: System Table Blocks on Load
### Parent Issue

_No response_

### Problem Statement

When a loaded site comes up, the System table blocks all calls to Config, which renders the server inaccessible.  See this thread dump:

https://gist.github.com/wezell/5bb3393d5b1b2cde7ad246a6315f7e6b

The only way to prevent the instances from falling over is to set:
```
          - name: DOT_ENABLE_SYSTEM_TABLE_CONFIG_SOURCE
            value: 'false'
```

### Steps to Reproduce

Maybe load a site with graphql queries and flush cache?

### Acceptance Criteria

We need to not ever block getting a config value.

### dotCMS Version

23.10

### Proposed Objective

Cloud Engineering

### Proposed Priority

Priority 1 - Show Stopper

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jdotcms**
removing the look for the system cache look up on the system table

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/business/CacheableEagerFactory.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    /**
+     * Finds all records and loads them into the cache
+     * @return Map<K, V>
+     * @throws DotDataException
+     */
+    default Map<K, V> findAll() throws DotDataException {
+
+        final Map<K, V> records = new LinkedHashMap<>();
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/business/CacheableEagerFactory.java:115]: Why a LinkedHashMap?  Why a TreeMap?

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/business/CacheableEagerFactory.java:115]: I think it was a mistake good catch

File: dotcms-integration/src/test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java [hunk 2]
```diff
@@ -933,6 +934,7 @@ public void Test_Vanity_URI_Ending_With_Forward_Slash_Handles_Non_forward_Slash_
      * @throws DotSecurityException
      * @throws DotDataException
      */
+    @Ignore  // this test is not valid anymore since now all contents are saved as json
```
Comment (wezell) [inline dotcms-integration/src/test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java]: delete it

**Review by wezell — CHANGES_REQUESTED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

**Review by wezell — APPROVED** [review]
looks good

**Pull request merged**